File: Data Structures & Algorithms/majority-element/submission-8.java
class Solution {
    public int majorityElement(int[] nums) {
        int res = 0;
        int count = 0;

        for(int n : nums){
            if (count == 0){
                res = n;
            }
            if ( n == res){
                count++;
            }
            else{
                count--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-8.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 0
        l = 0
        count = [0] * 26

        for r in range(len(s)):
            count[ord(s[r]) - 65] += 1
            while (r - l + 1) - max(count) > k:
                count[ord(s[l]) - 65] -= 1
                l += 1
            res = max(res, r - l + 1)
        return res 